src/ngx_http_lua_util.c

```diff
@@ -756,6 +767,57 @@ ngx_http_lua_inject_ngx_api(lua_State *L, ngx_http_lua_main_conf_t *lmcf,
     lua_setglobal(L, "ngx");
 
     ngx_http_lua_inject_coroutine_api(log, L);
+
+#ifdef OPENRESTY_LUAJIT
+    {
+        const char buf[] =
+            "local ngx_log = ngx.log\n"
+            "local ngx_WARN = ngx.WARN\n"
+            "local type = type\n"
+            "local ngx_get_phase = ngx.get_phase\n"
+            "local traceback = require 'debug'.traceback\n"
+            "local function newindex(table, key, value)\n"
+                "rawset(table, key, value)\n"
+                "local phase = ngx_get_phase()\n"
+                "if phase == 'init_worker' or phase == 'init' then\n"
+                    "return\n"
+                "end\n"
+                "local key_type = type(key)\n"
+                "if key_type == 'string' then\n"
+                    "ngx_log(ngx_WARN, 'lua http setting global variable, "
+                                       "key: ', key, "
+                                       "', value type: ', type(value), '\\n', "
+                                       "traceback())\n"
+                "else\n"
+                    "ngx_log(ngx_WARN, 'lua http setting global variable, "
+                                       "key type: ', key_type, "
+                                       "', value type: ', type(value), '\\n', "
+                                       "traceback())\n"
```

Comment:
Like above, we should log something like this:

```
lua http attempting to write to the global variable 'table: 0x40e6ca60'
```

So we can actually merge these 2 code branches and just use `tostring(key)` instead. It works on both the string-typed variable name and other types.